src/main/java/net/ccbluex/liquidbounce/features/module/modules/combat/SuperKnockback.kt

```diff
@@ -7,32 +7,109 @@ package net.ccbluex.liquidbounce.features.module.modules.combat
+    private val onlyMoveValue = BoolValue("OnlyMove", true)
+    private val onlyMoveForwardValue = object : BoolValue("OnlyMoveForward", true) {
+        override fun isSupported() = onlyMoveValue.get()
+    }
+
+    private var ticks = 0
+
+    val timer = MSTimer()
 
     @EventTarget
     fun onAttack(event: AttackEvent) {
         if (event.targetEntity is EntityLivingBase) {
-            if (event.targetEntity.hurtTime > hurtTimeValue.get())
+            if (event.targetEntity.hurtTime > hurtTimeValue.get() || !timer.hasTimePassed(delayValue.get().toLong()) ||
+                    (!MovementUtils.isMoving && onlyMoveValue.get()) || (!mc.thePlayer.onGround && onlyGroundValue.get())) {
+                return
+            }
+
+            if (onlyMoveForwardValue.get() && RotationUtils.getRotationDifference(Rotation(MovementUtils.movingYaw, mc.thePlayer.rotationPitch), Rotation(mc.thePlayer.rotationYaw, mc.thePlayer.rotationPitch)) > 35) {
                 return
+            }
+
+            when (modeValue.get().lowercase()) {
+
+                "legit" -> {
+                    ticks = 2
+                }
+
+                "silent" -> {
+                    ticks = 1
+                }
 
-            if (mc.thePlayer.isSprinting)
-                sendPacket(C0BPacketEntityAction(mc.thePlayer, C0BPacketEntityAction.Action.STOP_SPRINTING))
+                "packet" -> {
+                    if (mc.thePlayer.isSprinting) {
+                        mc.thePlayer.isSprinting = true
+                    }
+                    mc.netHandler.addToSendQueue(C0BPacketEntityAction(mc.thePlayer, C0BPacketEntityAction.Action.STOP_SPRINTING))
+                    mc.netHandler.addToSendQueue(C0BPacketEntityAction(mc.thePlayer, C0BPacketEntityAction.Action.START_SPRINTING))
+                    mc.thePlayer.serverSprintState = true
+                }
 
-            sendPacket(C0BPacketEntityAction(mc.thePlayer, C0BPacketEntityAction.Action.START_SPRINTING))
-            sendPacket(C0BPacketEntityAction(mc.thePlayer, C0BPacketEntityAction.Action.STOP_SPRINTING))
-            sendPacket(C0BPacketEntityAction(mc.thePlayer, C0BPacketEntityAction.Action.START_SPRINTING))
-            mc.thePlayer.isSprinting = true
-            mc.thePlayer.serverSprintState = true
+                "sneakpacket" -> {
+                    if (mc.thePlayer.isSprinting) {
+                        mc.thePlayer.isSprinting = true
+                    }
+                    mc.netHandler.addToSendQueue(C0BPacketEntityAction(mc.thePlayer, C0BPacketEntityAction.Action.STOP_SPRINTING))
+                    mc.netHandler.addToSendQueue(C0BPacketEntityAction(mc.thePlayer, C0BPacketEntityAction.Action.START_SNEAKING))
+                    mc.netHandler.addToSendQueue(C0BPacketEntityAction(mc.thePlayer, C0BPacketEntityAction.Action.START_SPRINTING))
+                    mc.netHandler.addToSendQueue(C0BPacketEntityAction(mc.thePlayer, C0BPacketEntityAction.Action.STOP_SNEAKING))
```

Comment:
If we have sendPacket, use it please. Thanks. Consistency.

Reply:
Please resolve this so it can be merged.

Reply:
in few hours I back my house and send!! Thanks ~ sorry

Reply:
already done